sycl/doc/SYCLPreprocessorMacroses.md

```diff
@@ -0,0 +1,20 @@
+# Overview
+
+This file describes macros that have effect on SYCL compiler and run-time.
+
+### RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR
+
+The spec assumes that the SYCL implementation does address space deduction. However,
+for our implementation, the deduction is performed in the middle end, where it's
+hard to provide user friendly diagnositcs.
+Due to these problems writing to raw pointers obtained from `constant_ptr` is not
+diagnosed now.
+The user can enable diagnostics upon writing to such pointers via enabling the
+`RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR` macro.
```

Comment:
The LIT tests don't seem to be covering this. Is this intentional?

Reply:
Diagnostics will point to SYCL headers, I don't know the way how to validate diagnostics pointing not to test file. If you know, we can probably add it. But, I'm not sure that we have to validate standard c++ diagnostics because this patch just re-uses them.

Reply:
Thanks, got it.